src/logic/addressBook/utils/index.ts

```diff
@@ -27,12 +32,15 @@ export const getNameFromAdbk = (addressBook, userAddress) => {
   return null
 }
 
-export const getOwnersWithNameFromAddressBook = (addressBook, ownerList) => {
+export const getOwnersWithNameFromAddressBook = (
+  addressBook: AddressBookEntryProps,
+  ownerList: Array<OwnerListEntryProps>,
+): Array<OwnerListEntryProps> | [] => {
   if (!ownerList) {
     return []
   }
   const ownersListWithAdbkNames = ownerList.map((owner) => {
-    const ownerName = getNameFromAdbk(addressBook, owner.address)
+    const ownerName = addressBook && getNameFromAdbk(addressBook, owner.address)
```

Comment:
1. You are checking that addressBook is defined, but according to the type it is always defined
2. Could you maybe check why this function gets an undefined address book? To me this solution looks like dealing with consequences rather than fix for an actual problem

Reply:
I've been investigated about the root cause of the problem and it seems to be a race condition one due to redux behaviour (when it renders for the first time, it may not have been completed the fetch of the address book). As a race condition problem (not predictable at all), I consider it as a good solution to mitigate the problem. This way, if in the first render, addressBook prop is undefined, we ensure that once it is defined, it will be handled properly without any error.

Reply:
Can't we just ensure that in this case an empty addressbook is returned instead of undefined?

Reply:
```
export const getAddressBook = createSelector(
  addressBookMapSelector,
  safeParamAddressFromStateSelector,
  (addressBook, safeAddress) => {
    return addressBook.get(safeAddress, List([]))
  },
)
```
making use of notSetValue param: https://immutable-js.github.io/immutable-js/docs/#/Map/get

Reply:
also you may omit the types here, I'm working on it as part of https://github.com/gnosis/safe-react/pull/1118